modules/javafx.controls/src/test/java/test/javafx/scene/control/TableCellTest.java

```diff
@@ -366,6 +366,25 @@ public void testCellInUneditableRowIsNotEditable() {
         assertFalse(cell.isEditing());
     }
 
+    @Test
+    public void testRowIsNotNullWhenAutoSizing() {
+        TableColumn<String, String> tableColumn = new TableColumn<>();
+        tableColumn.setCellFactory(col -> new TableCell<>() {
+
+            @Override
+            protected void updateItem(String item, boolean empty) {
+                super.updateItem(item, empty);
+
+                assertNotNull(getTableRow());
+            }
```

Comment:
feeling slightly uncomfortable with the generality of this: looks a bit like it's guaranteed that tableRow != null always (bullet 2 of our previous conversation) - would suggest to make it more specific to auto-sizing (f.i. start without auto-size, then trigger autosizing by code). Or at least doc it (add a message to the assertion, add the bug id) so that future readers will know what exactly is tested here :)

Reply:
Pretty sure table row is never null. Or is it on some corner case? Because even on an empty table, table rows with empty cells will have the row (since rows will be added in the virtualflow)

Reply:
> Pretty sure table row is never null. Or is it on some corner case? 

updateItem has no precondition :) So a clean implementation must cope with whatever state the cell is in (getting away with not thinking of potential corner cases most of the time).